src/v/raft/tests/persisted_stm_test.cc

```diff
@@ -766,3 +774,85 @@ TEST_F_CORO(state_machine_fixture, test_concurrent_apply_and_snapshot) {
     co_await std::move(write_sleep_f);
     co_await std::move(local_snapshot_f);
 }
+
+/**
+ * Tests the scenario in which an STM is added to the partition after it was
+ * already alive and Raft snapshot was taken on the partition.
```

Comment:
I'm not sure if I understand the scenario. The code adds both STMs at the same time before taking any raft snapshots.